app/models/spree/user.rb

```diff
@@ -156,9 +156,9 @@ def can_manage_orders?
     # any of order distributors allow them to edit their orders.
     def can_manage_line_items_in_orders?
       @can_manage_line_items_in_orders ||= begin
-        has_any_producer = enterprises.any?(&:is_producer)
-        has_producer_editable_orders = Spree::Order.editable_by_producers(enterprises).exists?
-        has_any_producer && has_producer_editable_orders
+        return unless enterprises.any?(&:is_producer_only)
+
+        Spree::Order.editable_by_producers(enterprises).exists?
       end
```

Comment:
I just noticed that a nil/false value would prevent caching of the result.

In this case, to cache the result we'd need to first check if the instance variable is defined and return it, for example: https://github.com/openfoodfoundation/openfoodnetwork/blob/master/app/controllers/spree/orders_controller.rb#L163

Reply:
We can also simplify without the block I think. If not `enterprises.any?`, the second condition doesn't get evaluated.

```suggestion
      return @can_manage_line_items_in_orders if defined? @can_manage_line_items_in_orders
      
      @can_manage_line_items_in_orders = (enterprises.any?(&:is_producer_only) && 
        Spree::Order.editable_by_producers(enterprises).exists?)
```

Reply:
Thanks @dacook - It's addressed here: https://github.com/openfoodfoundation/openfoodnetwork/pull/13113/commits/93cef666aee8121d706c26ef531f2be873064442

Reply:
Sorry my suggestion had a rubocop offence! I'll fix up the commit in place